### PR TITLE
Plans on jp dashboard card

### LINKED ISSUE #18419: Plans on JP: Update Dashboard Card
See pc8eDl-Vv-p2

### PR DESCRIPTION
**Pull request by ravishanker**
Adds dashboard card for wpcom plans

Fixes #18419 

To test:

1. Launch Jetpack app
2. Ensure previous [domain card](https://github.com/wordpress-mobile/WordPress-Android/pull/18286 ) is shown
3. Go to **App Settings** (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
4. Tap **Debug Settings**
5. Find **dashboard_card_free_to_paid_plans** [under Remote features](https://github.com/wordpress-mobile/WordPress-Android/pull/18418)
6. Enable the flag, and restart the app (scroll down)
7. Verify **plans card** appears as shown above on dashboard
8. Verify **domain card** doesn't appear
9. Tap on overflow (three dots) menu
10. **Verify** it pops up menu with **Hide this** option
11. Tap on Hide this
12. **Verify**, the card gets hidden 

>** Note **
>Copy on the card could change later

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests and manual tests

3. What automated tests I added (or what prevented me from doing so)
Updated and added unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt [hunk 1]
```diff
@@ -59,6 +59,7 @@ sealed class MySiteCardAndItem(open val type: Type, open val activeQuickStartIte
         BLOGGING_PROMPT_CARD,
         PROMOTE_WITH_BLAZE_CARD,
         DASHBOARD_DOMAIN_CARD,
+        DASHBOARD_PLANS_CARD,
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt:62]: **Ignorable Suggestion:** What about `DASHBOARD_PLAN_WITH_DOMAIN_CARD`?  I was considering the possibility of adding a direct plan selling card in the future, so `DASHBOARD_PLANS_CARD` would be a more suitable name for that card.

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsBuilderTest.kt [hunk 2]
```diff
@@ -332,6 +338,9 @@ class CardsBuilderTest : BaseUnitTest() {
                 dashboardCardDomainBuilderParams = DashboardCardDomainBuilderParams(
                     isEligibleForDomainCard, mock(), mock(), mock()
                 ),
+                dashboardCardPlansBuilderParams = DashboardCardPlansBuilderParams(
+                    isEligibleForDomainCard, mock(), mock(), mock()
```
Comment (irfano) [inline WordPress/src/test/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsBuilderTest.kt]: We're using `isEligibleForDomainCard` in `DashboardCardPlansBuilderParams`. I'm aware that this parameter can be used for both builder params, but let's add `isEligibleForPlansCard` (_or `isEligibleForPlanWithDomainCard` if you apply my name change suggestion_) parameter to the `buildDashboardCards(...)` to resolve the confusion.

File: WordPress/src/main/res/menu/dashboard_card_plans_menu.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:id="@+id/dashboard_card_plans_menu_item_hide_this"
+        android:title="@string/dashboard_card_domain_menu_hide_this"
```
Comment (irfano) [inline WordPress/src/main/res/menu/dashboard_card_plans_menu.xml]: **Fix request:** `dashboard_card_domain_menu_hide_this` → `dashboard_card_plans_menu_hide_this`

File: WordPress/src/main/res/layout/dashboard_card_plans.xml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.google.android.material.card.MaterialCardView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
```
Comment (irfano) [inline WordPress/src/main/res/layout/dashboard_card_plans.xml]: `xmlns:tools="http://schemas.android.com/tools"` is unnecessary here. Let's remove it from here and also from `dashboard_card_domain.xml`.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteSourceManager.kt [hunk 5]
```diff
@@ -35,6 +36,7 @@ class MySiteSourceManager @Inject constructor(
     promoteWithBlazeCardSource: PromoteWithBlazeCardSource,
     private val selectedSiteRepository: SelectedSiteRepository,
     private val dashboardCardDomainSource: DashboardCardDomainSource,
+    private val plansCardDomainSource: PlansCardDomainSource,
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteSourceManager.kt]: private val is unnecessary for dashboardCardDomainSource and plansCardDomainSource. Let's remove them.

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsBuilderTest.kt
Comment (irfano) [inline WordPress/src/test/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsBuilderTest.kt:1]: **Suggestion:** We can create equivalent test cases of `given is not eligible for domain, when cards are built, then domain card is not built()`, `given is eligible for domain, when cards are built, then domain card is built()` for plans card.

**Review by irfano — APPROVED** [review]
Thanks for handling this quickly, @ravishanker! I'm approving but please check my comments before merging.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt (hunk 1 above)
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt:62]: Good point.  Did play with different names, found them mostly verbose or redundant decorations.  Was wondering if dashboard would ever have more than one card about plans!  Dashboard is already clamouring for space as such.  Probably consider renaming it or finding a suitable one when the need arises.

**Pull request merged**